### DISCUSSION TITLE
Do I need to authorize OneDrive account when I just need to sync SharePoint in a Docker container?

### DISCUSSION THREAD

**Discussion by everwisher** [abraunegg/onedrive#1996] [Q&A] [answered]
As per _[Docker](https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md)_, OneDrive authorization needs to be done the moment the client container gets started. However, we need another command `onedrive --get-O365-drive-id '<your site name to search>'` to authorize access to SharePoint as per _[SharePoint-Shared-Libraries](https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md)_. 

In my case, I set up the container just to sync with SharePoint, I interrupted the OneDrive authorization process using Ctrl+C, and did `docker exec -it onedrive onedrive --get-O365-drive-id <mysite>` to get myself authorized successfully. But this container seems not to be syncing automatically after container is restarted unless I did `docker exec -it onedrive onedrive --monitor` and the container only starts to consume some CPU and RAM.

However, `docker logs od_movies` shows only the first item as follows with nothing appended to it:
> Base Args: --monitor
> \# We are being verbose
> \# Adding --verbose
> \# Launching onedrive
> Using 'user' Config Dir: /onedrive/conf
> No user or system config file found, using application defaults
> Checking Application Version ...
> Initializing the OneDrive API ...
> Configuring Global Azure AD Endpoints
> Authorize this app visiting:
> 
> https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxc&> scope=Files.ReadWrite%20Files.ReadWrite.all%20Sites.Read.All%20Sites.ReadWrite.All%20offline_access&> response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient
> 
> Enter the response uri: 
> usermod: no changes
> Base Args: --monitor
> \# We are being verbose
> \# Adding --verbose
> \# Launching onedrive
> Using 'user' Config Dir: /onedrive/conf
> No user or system config file found, using application defaults
> Checking Application Version ...
> Initializing the OneDrive API ...
> Configuring Global Azure AD Endpoints
> Authorize this app visiting:
> 
> https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxc&> scope=Files.ReadWrite%20Files.ReadWrite.all%20Sites.Read.All%20Sites.ReadWrite.All%20offline_access&> response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient
> 
> Enter the response uri: 
> usermod: no changes
> Base Args: --monitor
> \# We are being verbose
> \# Adding --verbose
> \# Launching onedrive
> Using 'user' Config Dir: /onedrive/conf
> No user or system config file found, using application defaults
> Checking Application Version ...
> Initializing the OneDrive API ...
> Configuring Global Azure AD Endpoints
> Authorize this app visiting:
> 
> https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxc&> scope=Files.ReadWrite%20Files.ReadWrite.all%20Sites.Read.All%20Sites.ReadWrite.All%20offline_access&> response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient
> 
> Enter the response uri: 
> usermod: no changes
> Base Args: --monitor
> \# We are being verbose
> \# Adding --verbose
> \# Launching onedrive
> Using 'user' Config Dir: /onedrive/conf
> No user or system config file found, using application defaults
> Checking Application Version ...
> Initializing the OneDrive API ...
> Configuring Global Azure AD Endpoints
> Authorize this app visiting:
> 
> https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxc&> scope=Files.ReadWrite%20Files.ReadWrite.all%20Sites.Read.All%20Sites.ReadWrite.All%20offline_access&> response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient

Is this because that I skipped the OneDrive authorization? How can I adapt the configuration to my SharePoint-only use case with Docker?

**Comment by abraunegg** [comment, marked as answer]
@everwisher 
> Is this because that I skipped the OneDrive authorization? How can I adapt the configuration to my SharePoint-only use case with Docker?

Correct .. it is because you did not follow the correct process.

What you need to do is run the client on a totally separate machine to get all the SharePoint details needed to configure your Docker Containers .. dont try and get all the details from inside a Docker Container ... will just lead to frustration and failure.

Follow the documentation on starting and running the containers as well. This is the documented process that *works*

**Comment by everwisher** [comment]
But following the documentation leads me to register with my ODFB account. Do you mean that I should copy the configuration file from another machine that syncs with the SharePoint site already to the container to be synced with SP and then run `onedrive --resync` because of the change of `drive_id` before a final `onedrive --monitor` ?
________________________________

**Reply by abraunegg** [reply]
> But following the documentation leads me to register with my ODFB account.

Correct ... each Docker container needs to be authenticated and authorised from a security scope

> Do you mean that I should copy the configuration file from another machine that syncs with the SharePoint site already to the container to be synced with SP

If you copy 'refresh_token' and only this file to your Docker 'config' folder this will negate the need to 'authenticate' that new Docker Container

> and then run `onedrive --resync` because of the change of `drive_id` before a final `onedrive --monitor` ?

Do *not* do this ..... do not use `--resync` unless the application tells you to.

You are also still starting the Docker container and application in the incorrect manner using Docker .. please read the documentation and follow what the documentation says on how to run the Docker container
